components/NameTag/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import * as S from "./styled";
+
+interface NameTag {
+  id: number;
+  created_at: string;
+  name: string;
+  user_id: number;
+  favorite: boolean;
+  link: {
+    count: number;
+  };
+}
+
+interface NameTagProps {
+  datas: NameTag[];
+  selected?: string;
+  onClick: (name: string, linksId?: number) => void;
+}
+
+function NameTag({ datas, selected, onClick }: NameTagProps) {
+  return (
+    <S.TageList>
+      <S.TagItem $selected={selected === "전체"} onClick={() => onClick("전체")}>
+        전체
+      </S.TagItem>
+      {datas?.map((data) => (
+        <S.TagItem key={data.id} $selected={selected === data.name} onClick={() => onClick(data.name, data.id)}>
+          {data.name}
```

Comment:
```suggestion
      {datas?.map(({id, name}) => (
        <S.TagItem key={id} $selected={selected === name} onClick={() => onClick(name, id)}>
          {name}
```

datas 이름만 봐서는 어떤 데이터인지 예상하기 어려울것같아요. 좀더 구체적인 이름을 지어주는건 어떨까요?? :-) 

map 메서드의 매개변수 data도 모호한것같아요. 
오히려 구조분해 할당을 바로 해주는건 어떨까요?